Author: Andreas Tille <tille@debian.org>
Last-Update: Thu, 19 Oct 2017 14:58:04 +0200
Description: Add missing headers

--- a/src/qtbeads/main_window.cpp
+++ b/src/qtbeads/main_window.cpp
@@ -10,6 +10,12 @@
 #include <QMenu>
 #include <QMenuBar>
 #include <QFileDialog>
+#include <QLineEdit>
+#include <QMessageBox>
+#include <QInputDialog>
+#include <QStatusBar>
+#include <QPrintDialog>
+#include <QFontDialog>
 #include "main_window.h"
 #include "QCimg.h"
 #include "../config.h"
